samples/template-operator/README.md

```diff
@@ -1,27 +1,59 @@
 It utilizes the [kubebuilder](https://book.kubebuilder.io/) framework with some modifications to implement Kubernetes APIs for custom resource definitions (CRDs).
 Additionally, it hides Kubernetes boilerplate code to develop fast and efficient control loops in Go.
 
 ## Contents
+* [Understanding Module Development in Kyma](#understanding-module-development-in-kyma)
 * [Implementation](#implementation)
   * [Pre-requisites](#pre-requisites)
   * [Generate kubebuilder operator](#generate-kubebuilder-operator)
   * [Default (declarative) Reconciliation and Status handling](#default-declarative-reconciliation-and-status-handling)
   * [Custom Reconciliation and Status handling guidelines](#custom-reconciliation-and-status-handling-guidelines)
   * [Local testing](#local-testing)
 * [Bundling and installation](#bundling-and-installation)
-  * [Makefile structure](#makefile-structure)
-  * [Build module operator image](#build-module-operator-image)
+  * [Grafana dashboard for simplified Controller Observability](#grafana-dashboard-for-simplified-controller-observability)
+  * [RBAC](#rbac)
+  * [Build module operator image](#prepare--build-module-operator-image)
   * [Build and push your module to the registry](#build-and-push-your-module-to-the-registry)
-* [RBAC](#rbac)
-* [Grafana dashboard](#grafana-dashboard)
+* [Using your Module in the Lifecycle Manager Ecosystem](#using-your-module-in-the-lifecycle-manager-ecosystem)
+  * [Deploying Kyma Infrastructure Operators with `kyma alpha deploy`](#deploying-the-kyma-infrastructure-operators-with-kyma-alpha-deploy)
+  * [Deploying a `ModuleTemplate` into the control-plane](#deploying-a-moduletemplate-into-the-control-plane)
+  * [Debugging the Operator Ecosystem](#debugging-the-operator-ecosystem)
+  * [Registering your Module within the Control-Plane](#registering-your-module-within-the-control-plane)
 
+## Understanding Module Development in Kyma 
+
+Before going in-depth, make sure you are familiar with:
+
+- [Modularization in Kyma](https://github.com/kyma-project/community/tree/main/concepts/modularization)
+- [Operator Pattern in Kubernetes](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+
+This Guide serves as comprehensive Step-By-Step tutorial on how to properly create a module from scratch by using an operator that is installing a Helm Chart. 
+Note that while other approaches are encouraged, there is no dedicated guide available yet and these will follow with sufficient requests and adoption of Kyma modularization.
+
+Every Kyma Module using an Operator follows 5 basic Principles:
+
+- Declared as available for use in a release channel through the `ModuleTemplate` Custom Resource in the Control-Plane
+- Declared as desired State within the `Kyma` Custom Resource in Runtime or Control-Plane
```

Comment:
```suggestion
- Declared as desired state within the `Kyma` custom resource in Runtime or Control Plane
```

Reply:
Custom Resource is a kubernetes definition so I will leave it capitalized. I will change all occurrences from Control-Plane to `control-plane` and Runtime to `runtime`